app/controllers/sign_up/select_email_controller.rb

```diff
@@ -21,6 +21,8 @@ def create
           user_id: current_user.id,
           email: form_params[:selection],
         )
+
+        session[:sp_email] = EmailContext.new(current_user).last_sign_in_email_address.id
```

Comment:
can we call this `sp_email_id` to make ti clear it's an ID?
```suggestion
        session[:sp_email_id] = EmailContext.new(current_user).last_sign_in_email_address.id
```

Reply:
Yes that reads clearer.